src/main/java/com/example/onlinebookstore/service/BookServiceImpl.java

```diff
@@ -0,0 +1,27 @@
+package com.example.onlinebookstore.service;
+
+import com.example.onlinebookstore.model.Book;
+import com.example.onlinebookstore.repository.BookRepository;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class BookServiceImpl implements BookService {
+    private final BookRepository bookRepository;
+
+    @Autowired
+    public BookServiceImpl(BookRepository bookRepository) {
+        this.bookRepository = bookRepository;
+    }
```

Comment:
* Autowired is redundant above the constructor if it is a single one with all required params
* you can remove the constructor at all and use lomboks `@ReqParamsConstructor`